modules/camera/camera_win.cpp

```diff
@@ -29,32 +29,382 @@
+		(void)m_pBuffer->QueryInterface(IID_PPV_ARGS(&m_p2DBuffer));
+	}
+
+	~VideoBufferLock() {
+		UnlockBuffer();
+		SafeRelease(&m_pBuffer);
+		SafeRelease(&m_p2DBuffer);
+	}
+
+	//-------------------------------------------------------------------
+	// LockBuffer
+	//
+	// Locks the buffer. Returns a pointer to scan line 0 and returns the stride.
+	//
+	// The caller must provide the default stride as an input parameter, in case
+	// the buffer does not expose IMF2DBuffer. You can calculate the default stride
+	// from the media type.
+	//-------------------------------------------------------------------
+
+	HRESULT LockBuffer(
+			LONG lDefaultStride, // Minimum stride (with no padding).
+			DWORD dwHeightInPixels, // Height of the image, in pixels.
+			BYTE **ppbScanLine0, // Receives a pointer to the start of scan line 0.
+			LONG *plStride // Receives the actual stride.
+	) {
+		HRESULT hr = S_OK;
+
+		// Use the 2-D version if available.
+		if (m_p2DBuffer) {
+			hr = m_p2DBuffer->Lock2D(ppbScanLine0, plStride);
+		} else {
+			// Use non-2D version.
+			BYTE *pData = NULL;
+
+			hr = m_pBuffer->Lock(&pData, NULL, NULL);
+			if (SUCCEEDED(hr)) {
+				*plStride = lDefaultStride;
+				if (lDefaultStride < 0) {
+					// Bottom-up orientation. Return a pointer to the start of the
+					// last row *in memory* which is the top row of the image.
+					*ppbScanLine0 = pData + abs(lDefaultStride) * (dwHeightInPixels - 1);
+				} else {
+					// Top-down orientation. Return a pointer to the start of the
+					// buffer.
+					*ppbScanLine0 = pData;
+				}
+			}
+		}
+
+		m_bLocked = (SUCCEEDED(hr));
+
+		return hr;
+	}
+
+	//-------------------------------------------------------------------
+	// UnlockBuffer
+	//
+	// Unlocks the buffer. Called automatically by the destructor.
+	//-------------------------------------------------------------------
+
+	void UnlockBuffer() {
+		if (m_bLocked) {
+			if (m_p2DBuffer) {
+				(void)m_p2DBuffer->Unlock2D();
+			} else {
+				(void)m_pBuffer->Unlock();
+			}
+			m_bLocked = FALSE;
+		}
+	}
+
 private:
+	IMFMediaBuffer *m_pBuffer;
+	IMF2DBuffer *m_p2DBuffer;
+
+	BOOL m_bLocked;
+};
+
+__forceinline BYTE Clip(int clr) {
+	return (BYTE)(clr < 0 ? 0 : (clr > 255 ? 255 : clr));
+}
+
+__forceinline RGBQUAD ConvertYCrCbToRGB(
+		int y,
+		int cr,
+		int cb) {
+	RGBQUAD rgbq;
+
+	int c = y - 16;
+	int d = cb - 128;
+	int e = cr - 128;
+
+	rgbq.rgbRed = Clip((298 * c + 409 * e + 128) >> 8);
+	rgbq.rgbGreen = Clip((298 * c - 100 * d - 208 * e + 128) >> 8);
+	rgbq.rgbBlue = Clip((298 * c + 516 * d + 128) >> 8);
+
+	return rgbq;
+}
+
+void TransformImage_YUY2(
+		BYTE *pDest,
+		LONG lDestStride,
+		const BYTE *pSrc,
+		LONG lSrcStride,
+		DWORD dwWidthInPixels,
+		DWORD dwHeightInPixels) {
+	for (DWORD y = 0; y < dwHeightInPixels; y++) {
+		RGBQUAD *pDestPel = (RGBQUAD *)pDest;
+		WORD *pSrcPel = (WORD *)pSrc;
+
+		for (DWORD x = 0; x < dwWidthInPixels; x += 2) {
+			// Byte order is U0 Y0 V0 Y1
```

Comment:
```suggestion
			// Byte order is U0 Y0 V0 Y1.
```
Per the [comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide).